### PR TITLE
JBPM-6771: Information not available in ProcessEventListener

### PR DESCRIPTION
**Pull request by mcivantos-tribalyte**

### CODE REVIEW

File: jbpm-flow/src/main/java/org/jbpm/process/instance/ProcessInstance.java [hunk 1]
```diff
@@ -64,5 +65,9 @@
 	String getDeploymentId();
 	
 	void setDeploymentId(String deploymentId);
+	
+	Date getStartDate();
+	
+	void setStartDate(Date startDate);
```
Comment (mswiderski) [inline jbpm-flow/src/main/java/org/jbpm/process/instance/ProcessInstance.java]: setter is certainly not needed as it should never be possible to set it other than process engine itself.

File: jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java [hunk 2]
```diff
@@ -123,7 +123,8 @@ public ProcessInstance getProcessInstance(long id, boolean readOnly) {
                 ProcessInstanceInfo processInstanceInfo = (ProcessInstanceInfo) context.findProcessInstanceInfo( id );
                 if ( processInstanceInfo == null ) {
                     return null;
-                }                
+                }  
+                processInstance.setStartDate(processInstanceInfo.getStartDate());
```
Comment (mswiderski) [inline jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java]: I would rather set this start date in org.jbpm.persistence.processinstance.ProcessInstanceInfo.getProcessInstance(InternalKnowledgeRuntime, Environment, boolean) ... and I see you dod that in the next file so why this is needed?

File: jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java [hunk 3]
```diff
@@ -751,6 +753,14 @@ public String getCorrelationKey() {
     public void setCorrelationKey(String correlationKey) {
         this.correlationKey = correlationKey;
     }
+    
+    public Date getStartDate() {
+        return startDate;
+    }
+    
+    public void setStartDate(Date startDate) {
+        this.startDate = startDate;
```
Comment (mswiderski) [inline jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java]: here setter is fine though I would most likely rename it to internalSetStartDate so it's not directly seen as set* and in addition allow setting it only when it was not set yet.

**Comment by mcivantos-tribalyte** [comment]
@mswiderski All your changes requested are already done.

File: jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java [hunk 4]
```diff
@@ -751,6 +753,14 @@ public String getCorrelationKey() {
...
         this.correlationKey = correlationKey;
     }
+    
+    public Date getStartDate() {
+        return startDate;
+    }
+    
+    public void internalSetStartDate(Date startDate) {
+        this.startDate = startDate;
+    }
```
Comment (MarianMacik) [inline jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java]: "in addition allow setting it only when it was not set yet." option is still missing.

Comment (mcivantos-tribalyte) [reply jbpm-flow/src/main/java/org/jbpm/workflow/instance/impl/WorkflowProcessInstanceImpl.java]: Done

**Comment by mswiderski** [comment]
ok to test

**Pull request merged**